Common/SymbolRepr.cs

```diff
@@ -0,0 +1,184 @@
+                }
+
+                if (!_futuresMonthCodeLookup.ContainsKey(expirationMonthString))
+                {
+                    return null;
+                }
+
+                var expirationMonth = _futuresMonthCodeLookup[expirationMonthString];
+
+                return Tuple.Create(underlyingString, expirationYearShort, expirationMonth);
+            }
+            else
+            {
+                // XYZH6 format
+                var expirationYearString = ticker.Substring(ticker.Length - 1, 1);
+                var expirationMonthString = ticker.Substring(ticker.Length - 2, 1);
+                var underlyingString = ticker.Substring(0, ticker.Length - 2);
+
+                int expirationYearShort;
+
+                if (!int.TryParse(expirationYearString, out expirationYearShort))
+                {
+                    return null;
+                }
+
+                if (!_futuresMonthCodeLookup.ContainsKey(expirationMonthString))
+                {
+                    return null;
+                }
+
+                var expirationMonth = _futuresMonthCodeLookup[expirationMonthString];
+
+                return Tuple.Create(underlyingString, expirationYearShort, expirationMonth);
+            }
+        }
+
+        /// <summary>
+        /// Returns future symbol ticker from underlying and expiration date. Function can generate tickers of two formats: one and two digits year
+        /// </summary>
+        /// <param name="underlying">String underlying</param>
+        /// <param name="expiration">Expiration date</param>
+        /// <param name="twoDigitsYear">True if year should represented by two digits; False - one digit</param>
+        /// <returns></returns>
+        public static string GenerateFutureTicker(string underlying, DateTime expiration, bool doubleDigitsYear = true)
+        {
+            var year = doubleDigitsYear ? expiration.Year % 100 : expiration.Year % 10;
+            return string.Format("{0}{1}{2}", underlying, _futuresMonthLookup[expiration.Month], year);
+        }
+
+        /// <summary>
+        /// Returns option symbol ticker in accordance with OSI symbology
+        /// More information can be found at http://www.optionsclearing.com/components/docs/initiatives/symbology/symbology_initiative_v1_8.pdf
+        /// </summary>
+        /// <param name="underlying">Underlying string</param>
+        /// <param name="right">Option right</param>
+        /// <param name="strikePrice">Option strike</param>
+        /// <param name="expiration">Option expiration date</param>
+        /// <returns></returns>
+        public static string GenerateOptionTickerOSI(string underlying, OptionRight right, decimal strikePrice, DateTime expiration)
+        {
+            if (underlying.Length > 5) underlying += " ";
+            return string.Format("{0,-6}{1}{2}{3:00000000}", underlying, expiration.ToString(DateFormat.SixCharacter), right.ToString()[0], strikePrice * 1000m);
```

Comment:
Since we're checking `underlying.Length > 5` here than perhaps the check doesn't need to happen in `Symbol.CreateOption` as well right before we invoke this method